components/match2/wikis/ageofempires/opponent_custom.lua

```diff
@@ -0,0 +1,42 @@
+---
+-- @Liquipedia
+-- wiki=ageofempires
+-- page=Module:Opponent/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Table = require('Module:Table')
+local Opponent = require('Module:Opponent')
+local Abbreviation = require('Module:Abbreviation')
+local TeamTemplate = require('Module:TeamTemplate')
+
+
+local CustomOpponent = Table.deepCopy(Opponent)
+
+function CustomOpponent.resolve(opponent, date)
+	if opponent.type == Opponent.team then
+		opponent.template = TeamTemplate.resolve(opponent.template, date) or 'tbd'
+	elseif Opponent.typeIsParty(opponent.type) then
+		local PlayerExt = require('Module:Player/Ext')
+		for _, player in ipairs(opponent.players) do
+			if Opponent.playerIsTbd(player) then
+				player.displayName = Abbreviation.make('TBD', 'To be determined (or to be decided)')
+			else
+				if player.flag then
+					PlayerExt.populatePageName(player)
+				else
+					PlayerExt.syncPlayer(player)
+				end
+				if player.team then
+					player.team = TeamTemplate.resolve(player.team, date)
+				else
+					player.team = PlayerExt.syncTeam(player.pageName, nil)
+				end
+			end
+		end
+	end
+	return opponent
+end
```

Comment:
```suggestion
function CustomOpponent.resolve(opponent, date)
	opponent = Opponent.resolve(opponent, date)
	if Opponent.typeIsParty(opponent.type) then
		for _, player in ipairs(opponent.players) do
			if Opponent.playerIsTbd(player) then
				player.displayName = Abbreviation.make('TBD', 'To be determined (or to be decided)')
			elseif player.team then
				player.team = TeamTemplate.resolve(player.team, date)
			end
		end
	end
	return opponent
end
```

Reply:
Would this work? I haven't checked the details

Reply:
The flag sync could be removed by this, the team sync still needs to be done.
Will adapt it like this.

Reply:
See 2e43286273816153b97128da4f719c006937f072